tuf/tuf.go

```diff
@@ -173,6 +173,15 @@ func (tr *Repo) RemoveBaseKeys(role string, keyIDs ...string) error {
 	return nil
 }
 
+// GetAllRoles returns a list of all role entries for this TUF repo
+func (tr *Repo) GetAllRoles() ([]*data.Role, error) {
+	roles := tr.keysDB.GetAllRoles()
+	if len(roles) == 0 {
+		return nil, data.ErrNoRoles{}
```

Comment:
I'm fine either way, though it seemed odd to not have any roles in a TUF repo since we have the base roles.  If you have a preference for empty array, I can definitely change this :)